From 8a9ee28a6e5083f9d0d3a8b604b3939b7446505f Mon Sep 17 00:00:00 2001
From: Alexander Todorov <atodorov@redhat.com>
Date: Mon, 15 Jun 2015 09:49:46 -0400
Subject: [PATCH 12/43] tests: Fall back to C.UTF-8 if no en_US.utf8 available

Not all distributions (Fedora) have C.UTF-8 so first try en_US.utf8 and fall
back to it. This may still fail if neither exist.

* tests/t0251-gpt-unicode.sh: Test for en_US and fall back to C.UTF-8

Signed-off-by: Brian C. Lane <bcl@redhat.com>
---
 tests/t0251-gpt-unicode.sh | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/tests/t0251-gpt-unicode.sh b/tests/t0251-gpt-unicode.sh
index c845950..78ad4eb 100755
--- a/tests/t0251-gpt-unicode.sh
+++ b/tests/t0251-gpt-unicode.sh
@@ -22,7 +22,11 @@ dev=loop-file
 # create zeroed device
 truncate -s 10m $dev || fail=1
 
-export LC_ALL=C.UTF-8
+LC_ALL=$(locale -a | grep en_US.utf8)
+if [ -z "$LC_ALL" ]; then
+   LC_ALL=C.UTF-8
+fi
+export LC_ALL="$LC_ALL"
 # create gpt label with named partition
 part_name=$(printf 'foo\341\264\244')
 parted -s $dev mklabel gpt mkpart primary ext2 1MiB 2MiB name 1 $part_name > empty 2>&1 || fail=1
-- 
1.9.5

